pallets/clusters/src/lib.rs

```diff
@@ -441,10 +458,32 @@ pub mod pallet {
 
 			ensure!(<Roles<T>>::contains_key(&cluster_id, &name_index), Error::<T>::RoleNotFound);
 
+			let role_settings = settings
+				.into_inner()
+				.into_iter()
+				.map(|setting| RoleSetting {
+					name: Self::take_name_index(&setting.name),
+					data: setting.data,
+				})
+				.collect::<Vec<RoleSetting>>();
+
+			// ensure that the Role does not contain already any of the setting names provided
+			for setting in &role_settings {
+				ensure!(
+					!<Roles<T>>::get(&cluster_id, &name_index)
+						.ok_or(Error::<T>::SystematicFailure)?
+						.settings
+						.into_iter()
+						.map(|setting| setting.name)
+						.collect::<Vec<Compact<u64>>>()
```

Comment:
I'm sure you don't need to collect... just filter directly using `any` or `all`

Reply:
without considering that this is called get on same role repeatedly in the loop 🤦‍♂️

Reply:
also it seems there is no way to just change a setting currently. need to remove first.